File: frontend/projects/api/src/lib/models/today-reading-dto.ts
/**
 * The reading pillar's slice of `TodayDto`: the current day of the active
 * plan — the most recent day completed today, or else the first uncompleted
 * day.
 */
export interface TodayReadingDto {
  readonly dayId: string;
  readonly dayNumber: number;
  readonly passageReference: string;
  readonly youVersionUrl: string;
  readonly completedToday: boolean;
  readonly planName: string;
  readonly completedCount: number;
  readonly totalDays: number;
  /**
   * Tomorrow's likely passage for the evening glance — the first
   * still-uncompleted day (today's while it isn't done), or null when the
   * plan would be finished.
   */
  readonly nextPassageReference: string | null;
}

File: backend/src/Shalom.Application/Contracts/TodayReadingDto.cs
namespace Shalom.Application.Contracts;

/// <summary>
/// The reading pillar's slice of <see cref="TodayDto"/>: the current day of
/// the active plan. The current day is the most recent day completed today
/// (so the card shows its done state) or otherwise the first uncompleted day.
/// </summary>
public record TodayReadingDto(
    Guid DayId,
    int DayNumber,
    string PassageReference,
    string YouVersionUrl,
    bool CompletedToday,
    string PlanName,
    int CompletedCount,
    int TotalDays);
